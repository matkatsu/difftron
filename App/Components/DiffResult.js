// @flow
import React from 'react';
import AceEditor from 'react-ace';
import 'diff2html.global.css';
import styles from 'Components/Styles/DiffResult.css';
import DIFF_FORMAT from 'Config/DiffFormat';
import type { Contents, Language, Theme, Format, Split } from 'Types/State';

type EditorSettings = {
  format: Format,
  split: Split,
  language: Language,
  theme: Theme,
}

type Props = {
  value: Contents,
  editorSettings: EditorSettings,
  handleSplitChange: (split: Split) => void
};

const lineByLine: Split = 'line-by-line';
const sideBySide: Split = 'side-by-side';

const DiffResult = ({ value, editorSettings, handleSplitChange }: Props) => {
  let component;
  if (editorSettings.format === DIFF_FORMAT.html) {
    component = (
      <div>
        <div className={styles.splitFormArea}>
          <select value={editorSettings.split} onChange={handleSplitChange}>
            <option value={lineByLine}>Line by Line</option>
            <option value={sideBySide}>Side by Side</option>
          </select>
        </div>
        <div dangerouslySetInnerHTML={{ __html: value }} />
      </div>
    );
  } else {
    component = (
      <AceEditor
        mode={editorSettings.language}
        theme={editorSettings.theme}
        name="DIFF_RESULT"
        width="100%"
        height="100%"
        value={value}
        showPrintMargin={false}
        showGutter={false}
        editorProps={{ $blockScrolling: true }}
      />
    );
  }
  return component;
};

export default DiffResult;
