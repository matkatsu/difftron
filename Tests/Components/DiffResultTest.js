/* eslint-disable import/first */
import 'ignore-styles';
import '../Setup';
import test from 'ava';
import React from 'react';
import render from 'react-test-renderer';
import DiffResult from 'Components/DiffResult';

test('DiffResult component unified', (t) => {
  const props = {
    value: 'result contents',
    editorSettings: {
      format: 0,
      split: 'side-by-side',
      language: 'javascript',
      theme: 'github',
    },
    handleSplitChange: () => ({}),
  };
  const tree = render.create(<DiffResult {...props} />).toJSON();
  t.snapshot(tree);
});

test('DiffResult component html', (t) => {
  const props = {
    value: 'result contents',
    editorSettings: {
      format: 1,
      split: 'side-by-side',
      language: 'javascript',
      theme: 'github',
    },
    handleSplitChange: () => ({}),
  };
  const tree = render.create(<DiffResult {...props} />).toJSON();
  t.snapshot(tree);
});
