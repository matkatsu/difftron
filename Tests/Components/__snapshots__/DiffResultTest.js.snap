exports[`DiffResult component 1`] = `"{"__ava_react_jsx":{"type":"div","props":{"id":"DIFF_RESULT","style":{"width":"100%","height":"100%"}},"children":null}}"`;

exports[`DiffResult component html 1`] = `"{\"__ava_react_jsx\":{\"type\":\"div\",\"props\":{},\"children\":[{\"type\":\"div\",\"props\":{},\"children\":[{\"type\":\"select\",\"props\":{\"value\":\"side-by-side\"},\"children\":[{\"type\":\"option\",\"props\":{\"value\":\"line-by-line\"},\"children\":[\"Line by Line\"]},{\"type\":\"option\",\"props\":{\"value\":\"side-by-side\"},\"children\":[\"Side by Side\"]}]}]},{\"type\":\"div\",\"props\":{\"dangerouslySetInnerHTML\":{\"__html\":\"result contents\"}},\"children\":null}]}}"`;

exports[`DiffResult component unified 1`] = `"{\"__ava_react_jsx\":{\"type\":\"div\",\"props\":{\"id\":\"DIFF_RESULT\",\"style\":{\"width\":\"100%\",\"height\":\"100%\"}},\"children\":null}}"`;
